### PR TITLE
feat: add multi select field component

### PR DESCRIPTION
**Pull request by tibs245**
Résultat : 
startui-mutliselect

Thank to  @yoannfleurydev . I just copied his code 😂

### CODE REVIEW

File: src/components/FieldMultiSelect/index.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        id={id}
+        value={options?.filter((option) => value?.includes(option.value)) || []}
+        onBlur={() => setIsTouched(true)}
+        placeholder={placeholder}
+        onChange={handleChange}
+        options={options}
+        isDisabled={isDisabled}
+        isClearable={!isNotClearable}
+        isSearchable={isSearchable}
+        noOptionsMessage={() => noOptionsMessage || 'Aucune option'}
```
Comment (yoannfleurydev) [inline src/components/FieldMultiSelect/index.tsx:85]: Please add at least english and french translations for this part.

```suggestion
        noOptionsMessage={() => noOptionsMessage || t('somethinghere')}
```

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

File: src/components/FieldMultiSelect/docs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <Formiz autoForm connect={form}>
+      <FieldMultiSelect
+        name="mySelect"
+        label="Label"
+        helper="Helper"
+        options={options}
+      />
+
+      <Code mt={5}>{JSON.stringify(form.values, null, 3)}</Code>
```
Comment (ivan-dalmet) [inline src/components/FieldMultiSelect/docs.stories.tsx:26]:
```suggestion
      <Code mt={5}>{JSON.stringify(form.values, null, 2)}</Code>
```

File: src/components/FieldMultiSelect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from 'react';
```
Comment (ivan-dalmet) [inline src/components/FieldMultiSelect/index.tsx:1]: The export of this component is missing in the components/index.ts file ;)

Comment (tibs245) [reply src/components/FieldMultiSelect/index.tsx:1]: @ivan-dalmet I use addons to generate it. And I see many component not exported. 

You want I push all export in this MR or another MR ?

image

Comment (yoannfleurydev) [reply src/components/FieldMultiSelect/index.tsx:1]: Do it in this one please :)

Comment (tibs245) [reply src/components/FieldMultiSelect/index.tsx:1]: Done :+1:

File: src/components/FieldMultiSelect/index.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    showError,
+    ...rest,
+  };
+
+  const handleChange = (values) => {
+    if (!values || !values.length) {
+      setValue(null);
+      return;
+    }
+    setValue(values?.map((value) => value.value));
```
Comment (ivan-dalmet) [inline src/components/FieldMultiSelect/index.tsx:73]:
```suggestion
    setValue(values?.map((option) => option.value));
```

**Comment by yoannfleurydev** [comment]
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=177&id=BearStudio_start-ui&open=AX8wN1kBAroeSt1grC3X

File: src/components/FieldMultiSelect/index.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    errorMessage,
+    helper,
+    id,
+    isRequired: !!required,
+    label,
+    showError,
+    ...rest,
+  };
+
+  const handleChange = (optionsSelectionned) => {
```
Comment (ivan-dalmet) [inline src/components/FieldMultiSelect/index.tsx:68]:
```suggestion
  const handleChange = (optionsSelected) => {
```

File: src/components/FieldMultiSelect/index.tsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    if (!optionsSelectionned || !optionsSelectionned.length) {
```
Comment (ivan-dalmet) [inline src/components/FieldMultiSelect/index.tsx:69]:
```suggestion
    if (!optionsSelected?.length) {
```

File: src/components/FieldMultiSelect/index.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      setValue(null);
+      return;
+    }
+    setValue(optionsSelectionned?.map((option) => option.value));
```
Comment (ivan-dalmet) [inline src/components/FieldMultiSelect/index.tsx:73]:
```suggestion
    setValue(optionsSelected?.map((option) => option.value));
```

**Review by ivan-dalmet — CHANGES_REQUESTED** [review]

File: src/components/FieldMultiSelect/docs.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { Code } from '@chakra-ui/react';
+import { Formiz, useForm } from '@formiz/core';
+
+import { FieldMultiSelect } from './index';
```
Comment (yoannfleurydev) [inline src/components/FieldMultiSelect/docs.stories.tsx:4]:
```suggestion
import { FieldMultiSelect } from '.';
```

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

File: src/components/FieldMultiSelect/docs.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+export default {
+  title: 'Fields/FieldMultiSelect',
+};
+export const Default = () => {
+  const form = useForm({ subscribe: true });
+
+  const options = [
+    { label: 'One', value: 'One' },
+    { label: 'two', value: 'Two' },
```
Comment (yoannfleurydev) [inline src/components/FieldMultiSelect/docs.stories.tsx:14]:
```suggestion
    { label: 'Two', value: 'Two' },
```

**Review by yoannfleurydev — APPROVED** [review]

**Review by ivan-dalmet — APPROVED** [review]

**Pull request merged**